src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/FSharpCoreReflectionProxy.cs

```diff
@@ -204,15 +204,18 @@ public Func<IEnumerable<Tuple<TKey, TValue>>, TFSharpMap> CreateFSharpMapConstru
         }
 
         private Attribute? GetFSharpCompilationMappingAttribute(Type type)
-            => type.GetCustomAttribute(_compilationMappingAttributeType, inherit: true);
+        {
+            object[] attributes = type.GetCustomAttributes(_compilationMappingAttributeType, inherit: false);
+            return attributes.Length == 0 ? null : (Attribute)attributes[0];
```

Comment:
```suggestion
            return attributes is [Attribute attr, ..] ? attr : null;
```

Reply:
I believe this won't work in netstandard2.0 either because it's missing the necessary polyfills.